app/src/main/java/com/framgia/toeic/data/source/local/VocabularyLessonData.java

```diff
@@ -0,0 +1,44 @@
+package com.framgia.toeic.data.source.local;
+
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+
+import com.framgia.toeic.data.VocabularyLessonDataSource;
+import com.framgia.toeic.data.model.VocabularyLesson;
+import com.framgia.toeic.data.source.CallBack;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class VocabularyLessonData implements VocabularyLessonDataSource.Local {
+    private static final String TABLE_lESSON_VOCABULARY = "tbl_lesson_vocabulary";
+    private static final String COLUMN_ID = "id";
+    private static final String COLUMN_NAME = "name";
+    private DBHelper mDBHelper;
+
+    public VocabularyLessonData(DBHelper dbHelper) {
+        this.mDBHelper = dbHelper;
+    }
+
+    @Override
+    public void getVocabulary(CallBack<List<VocabularyLesson>> callBack) {
+        try {
+            mDBHelper.createDataBase();
+        } catch (IOException e) {
+            throw new Error(e.getMessage());
+        }
```

Comment:
if some exception appeared, please return by a callback 
and finish this function by return.